src/ba.py

```diff
@@ -34,4 +47,32 @@ def strict_border_array(x: str) -> list[int]:
     >>> strict_border_array("")
     []
     """
-    return []  # FIXME
+
+    if len(x) == 0:
+        return []
+
+    else:
+        x += '$'
+        bax = (len(x)-1)*[0]
+        for i in range(len(x)-1):
+            print(bax)
+            if x[i+1] == x[0]:
+                o = 1
+                while x[i+1+o] == x[o]: # extendable
+                    o+=1
+                    if x[i+1+o] != x[o]:
+                        bax[i+o] = o
+                        break
+                else:
+                    bax[i+o] = max(o, bax[i+o])
```

Comment:
We have an inner loop here. Are we sure that it runs a constant number of times on average? Otherwise, after all, we don't have a linear time algorithm here. At the top of my head, I can't work out exactly what is happening in the code, so I can't tell. Luckily, you hold the burden of proof, so I don't have to ;)